src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -13098,15 +13124,80 @@ static class E
     }
 }
 """;
-        try
-        {
-            // Tracked by https://github.com/dotnet/roslyn/issues/78828 : assertion in NullableWalker
-            var comp = CreateCompilation(src);
-            CompileAndVerify(comp, expectedOutput: "42").VerifyDiagnostics();
-        }
-        catch (InvalidOperationException)
-        {
-        }
+        var comp = CreateCompilation(src);
+        CompileAndVerify(comp, expectedOutput: "42").VerifyDiagnostics();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/78828")]
+    public void ExtensionGetEnumerator_NullReceiver()
+    {
+        var src = """
+            #nullable enable
+            C? c = null;
+            foreach (var x in c) // 1
+            {
+                System.Console.Write(x);
+                break;
+            }
+
+            class C
+            {
+                public bool MoveNext() => true;
+                public int Current => 42;
+            }
+
+            static class E
+            {
+                extension(C c)
+                {
+                    public System.Collections.Generic.IEnumerator<int> GetEnumerator() => throw null!;
+                }
+            }
+            """;
+        var comp = CreateCompilation(src);
+        // Tracked by https://github.com/dotnet/roslyn/issues/78830 : diagnostic quality consider reporting a better containing symbol
+        comp.VerifyEmitDiagnostics(
+            // (3,19): warning CS8604: Possible null reference argument for parameter 'c' in 'extension(C)'.
+            // foreach (var x in c) // 1
+            Diagnostic(ErrorCode.WRN_NullReferenceArgument, "c").WithArguments("c", "extension(C)").WithLocation(3, 19));
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/78828")]
+    public void ExtensionGetEnumerator_Reinference()
+    {
+        var src = """
+            #nullable enable
+            var s = "a";
+            if (string.Empty == "")
+                s = null;
+
+            var c = M(s); // 'C<string?>'
+            foreach (var x in c)
```

Comment:
nit: Consider also covering the happy case where `IEnumerator<string>` is inferred/returned